### PR TITLE
Add Generation Presets content to docs

### PR DESCRIPTION
**Pull request by pwoznic**
Added Generation presets to Query API Definition, API Reference, and some misc updates around response languages and FCS.

### CODE REVIEW

File: www/docs/api-reference/generation-presets/list-generation-presets.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+over query responses. This includes the `prompt_template`, the Large Language 
+Model (LLM), and other generation settings like `max_tokens` and `temperature`. 
+Users specify a generation preset in their query or chat requests using the 
+`generation-preset-name` field.
+
+
+
+:::note
+
+The `generation-preset-name` field replaces the `prompt_name` field that was in the
```
Comment (nikron) [inline www/docs/api-reference/generation-presets/list-generation-presets.md:34]: generation_preset_name

File: www/docs/api-reference/generation-presets/list-generation-presets.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+
+
+:::note
+
+The `generation-preset-name` field replaces the `prompt_name` field that was in the 
+`generation` object of a `query`. Existing queries using the previous 
+`prompt_name` field is now deprecated but still supported for backward 
+compatibility. 
+work
```
Comment (nikron) [inline www/docs/api-reference/generation-presets/list-generation-presets.md:38]: trailing word

File: www/docs/api-reference/search-apis/search.md [hunk 3]
```diff
@@ -51,8 +57,8 @@ the body that specifies the following:
 - `stream_response` - Indicates whether to stream the response in real-time (`true`) or
   to send a complete summary at the end of processing the request (`false`)
 - `search` - Specifies the search parameters
-- `generation` - Specifies the summarization parameters. Excluding this generation 
-  field disables summarization.
+- `generation` - Specifies the summarization parameters and generation presets.
```
Comment (nikron) [inline www/docs/api-reference/search-apis/search.md:60]: and the generation preset?

File: www/docs/api-reference/search-apis/search.md [hunk 4]
```diff
@@ -313,22 +358,38 @@ To disable summarization, exclude the `generation` object from a query.
 capabilities, which present a powerful toolkit for tailoring summarizations to
 specific application and user needs.
 
-The `prompt_name` allows you to specify one of our [available summarizers](/docs/learn/grounded-generation/select-a-summarizer).
-Use `prompt_name` and `prompt_text` to override the default prompt with a
+If you need more customization beyond what is available in the presets,
```
Comment (nikron) [inline www/docs/api-reference/search-apis/search.md:361]: if you need to change generation beyond what the preset sets?

something like that

also only scale customers can do the majority of preset overries at this piont

File: www/docs/api-reference/generation-presets/list-generation-presets.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+import CodeBlock from '@theme/CodeBlock';
+import {vars} from '@site/static/variables.json';
+import {Config} from '@site/docs/definitions.md';
+
+Organizations often struggle to fine-tune query responses and maintain 
+consistency across different use cases. Vectara creates and maintains 
+predefined generation presets for our users which provides a flexible and 
+powerful way to utilize generation parameters. Each preset includes a 
+complete Velocity template for the prompt along with other generation 
+parameters. Presets are typically associated with specific LLMs, though some
```
Comment (nikron) [inline www/docs/api-reference/generation-presets/list-generation-presets.md:18]: a preset is currently associated with a single llm at this time.

**Review by nikron — CHANGES_REQUESTED** [review]

**Comment by pwoznic** [comment]
I think I addressed them all :)

**Comment by tallatshafaat** [comment]
The list-generation-presets API reference is appearing weirdly: https://deploy-preview-318--luxury-shortbread-acee05.netlify.app/docs/rest-api/list-generation-presets

**Pull request merged**